ccgateway/Program.cs

```diff
@@ -27,20 +27,118 @@
 using Microsoft.Extensions.Configuration;
 using System.Diagnostics;
 using System.Text;
+using System.Threading.Tasks;
+using CCGatewayPlugin;
+using System.Threading;
 
 namespace ccgateway
 {
     class Program
     {
+        static string folder;
+        static IConfiguration config;
+        static async Task ProcessRequest(Loader<ICCGatewayPluginAsync> loader, NetMQFrame id, string requestString)
+        {
+            string[] command = requestString.Split();
+            string response;
+
+            if (command.Length < 2)
+            {
+                response = "poor";
+                Console.WriteLine(requestString + ": not enough parameters");
+            }
+            else
+            {
+                string action = command[0];
+                command = command.Skip(1).ToArray();
+
+                ICCGatewayPluginAsync plugin = loader.Get(action);
+                var pluginConfig = config.GetSection(action);
```

Comment:
Suggestion: you can move L53 and L56 inside the else block on L61 b/c they aren't needed outside of it (shaving off a few microseconds probably).